### ISSUE TITLE
[UX] Change user/login to /login and /user/logout to /logout

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1248]
Because it would just make so much more sense to the rest of the world.

**Comment by dariusgarza** [comment]
💜💛💚💙

**Comment by quicksketch** [comment]
Like other rename X to Y issues that move paths, we can do this as long as we set up a redirect from `user/login` to `/login` for existing sites.

**Comment by klonos** [comment]
:+1: and @quicksketch was about to suggest the very same thing. You beat me to it.

**Comment by BWPanda** [comment]
Here's a [PR](https://github.com/backdrop/backdrop/pull/2848). Be interested to see if all tests pass on first try... :wink:

**Comment by findlabnet** [comment]
1) Seems like half of solution.
What about 'user/register' and 'user/password'?
2) IMHO, path 'user/*' is very useful as first level check for security vulnerabilities like 'drupalgeddon'.

**Comment by BWPanda** [comment]
@findlabnet Yes, I noticed `user/register` and `user/password` when I was making this PR. I thought to keep the PR as requested above for now and see what feedback comes through, but I think that if this goes ahead then we should think about changing those URLs as well.

Another thought I had but forgot to mention; it'd be much simpler/easier to just create URL aliases by default for `'login' -> 'user/login'`, so that's another option worth considering maybe...

**Comment by klonos** [comment]
There's a couple of failing tests @BWPanda 😄 

I don't think that we should be relying on the core Redirect module for the redirects, as that module may be disabled, or never installed for some profiles/distributions. So this will be more prone to breakages. I wonder how previous redirects were made in similar situations, but my hunch tells me most likely something to do with the menu/routes system (could be wrong though), or plain simple `backdrop_goto()`.

There's at least one instance where `user/login` should not be changed, as it is there for backwards compatibility (left a comment in the PR), and there are a lot of instances in docblocks where previously `user/login` made sense, whereas I think that we now need to specify that we are referring to a path. For example:

> ... This provides the flexibility to build a **link to user/login** and override the default redirection so that the user is redirected to a specific path after logging in:

...has been changed to this:

> ...This provides the flexibility to build a **link to login** and override the default redirection so that the user is redirected to a specific path after logging in:

...but should better be:

> ...This provides the flexibility to build a **link to the "login" path**, and override the default redirection, so that the user is redirected to a specific path after logging in:

**Comment by BWPanda** [comment]
I was looking at updating my PR here, but then I thought of something...

What if there's a site that has setup a custom page/layout/view/etc. at the path `/login`? This issue change would conflict/overwrite that, right?

Is there a way to make this change only for new sites? If not changing every single instance, what about just setting up the new `/login` path and then a redirect from `/user/login`?

If not possible to do this just for new sites, then this might need to be a 2.x thing...

**Comment by jlfranklin** [comment]
Why don't we create a contrib module with `hook_menu_alter()` to change the paths.  This would be a good way to gauge interest in this change with the community.

**Comment by bugfolder** [comment]
I was going to go test https://github.com/backdrop/backdrop/pull/2848, but reading through the comments (and having some further thoughts of my own) makes me think more feedback is needed. It seems like there's several things to consider:

* Just change `user/login` and `user/logout`
* Also change `user/register` and `user/password`
* Any of the above, but also search for existing paths, aliases or redirects for the new paths (and then what? Don't make the change if there's something in the way?)
* Don't change any of the above, but add redirects from the simpler paths (if there's not existing paths, aliases, or redirects)

**Comment by findlabnet** [comment]
Some contribs (at least "Login allowlist/Login whitelist") use or can handle the "user/" path as an entry point. Not a big deal, but it is.
